src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -937,19 +943,25 @@ internal void DoGraphAutoLayout()
             // Check if all the selected models are groups
             bool isGroupLayout = selection.Count > 0 &&
                 selection.All(x => x is AnnotationModel ||
-                    selection.Where(g => g is AnnotationModel)
-                    .Any(g => (g as AnnotationModel).SelectedModels.Contains(x)));
+                    selection.OfType<AnnotationModel>().Any(g => g.SelectedModels.Contains(x)));
 
             GenerateCombinedGraph(isGroupLayout);
             RecordUndoGraphLayout(isGroupLayout);
-            GenerateSeparateSubgraphs();
+
+            // Generate subgraphs separately for each cluster
+            SubgraphClusters.ForEach(
```

Comment:
If several groups are selected, then each group's nodes are marked as one cluster. Otherwise, all nodes are in one overall cluster. Then, subgraphs are generated separately for each cluster.